### DISCUSSION TITLE
Why do you need Wayland support ?

### DISCUSSION THREAD

**Discussion by SuperUserNameMan** [raysan5/raylib#4174] [Q&A]
Hello,

We're currently trying to fix some bugs in #4151 on every GLFW desktop platforms, which include Wayland.

So far we made progresses with MS Windows, Linux+X11, and MacOS.

However, the Linux+Wayland is another story, because Wayland does not allow to know or set the position of a window.

The consequence is that it breaks everything that relies on these data, including some basic functions like `GetCurrentMonitor()`.

So, i'm wondering, why do you need Wayland support for exactly ?

I'm asking, because if i understood correctly, Wayland offers an X11 backward compatibility extension (at least on Ubuntu).

So, you compile once for X11, and your binary can run on every X11 or Wayland Linuxes.

So why do you need to compile Raylib with `GLFW_LINUX_ENABLE_X11=FALSE` and `GLFW_LINUX_ENABLE_WAYLAND=TRUE` ??

Is there a non-niche use case scenario that i'm not aware of where Wayland must absolutely work without the X11 compatibility extension ?

**Comment by ghost** [comment]


**Reply by SuperUserNameMan** [reply]
> They have a very good reason for it. Mir's doc had an excellent overview about it: https://mir-server.io/docs/window-positions-under-wayland

I'm not really convinced that removing the access to the window position was really a good idea if they don't provide alternatives to some some specific use case, like guessing on which monitor a window is mostly visible for instance.

> [xwayland](https://wayland.freedesktop.org/xserver.html) likely won't be viable forever.

Also, i'm noticing that Xwayland is not 100% reliable neither :-/

> But ignoring Wayland probably won't work on the long term.

Despite my skepticism and my desire to euthanize Wayland support, i managed to make the mentioned PR  somehow functional on Wayland today (a least on Ubuntu). Some workarounds and tricks were required.
So for now, Wayland support is a little crippled compared to other platforms, but safe ... 

Thanks for this enlightenment.

**Comment by CrackedPixel** [comment]
> I'm asking, because if i understood correctly, Wayland offers an X11 backward compatibility extension (at least on Ubuntu). So, you compile once for X11, and your binary can run on every X11 or Wayland Linuxes.

Wayland and X11 are very different and Wayland is NOT backwards compatible as X11 is not backwards to it. Most Linux users will use something like `xwayland` which can allow some x11 apps to run in Wayland environments but it is not perfect

Some users may want to target Wayland specifically and **raylib offers this choice to the developer.**

**Comment by wwderw** [comment]
As much as I'm not a fan of Wayland as it is now (I speak as more of an artist, Cintiq support is a regression compared to the x11 driver), I do think there is too much of a push for it and I don't see x11/xwayland being around forever and probably that is close to happening compared to other technology that is in the same boat, limping along with compatibility layers.

It is certainly not perfect with xwayland.  I see far more artifacting and just in general slow response times with some programs/games.

**Comment by irik77587** [comment]
One use case of wayland would be screen capture. Important for streamers but unnecessary hurdle for gamedevs.

**Comment by jestarray** [comment]
xwayland, e.g the compatability layer for x11 still has some issues, one of which is that it opens popup menus on the wrong monitor sometimes(last time i used it a year ago). Native I think might be better latency by not having to go through the x11 layer? don't quote me on that
